src/scos_tekrsa/hardware/tekrsa_sigan.py

```diff
@@ -286,6 +288,17 @@ def acquire_time_domain_samples(
         if gain_adjust:
             # Get calibration data for acquisition
             calibration_args = [self.sample_rate, self.frequency, self.reference_level]
+            if self.device_name in [
```

Comment:
It might be slightly more complicated, and would require changes in scos-actions, but if we modify acquire_time_domain_samples to take a dictionary of the parameters like the actions use then we could use the calibration.calibration_parameters to dynamically pull the appropriate values from the dictionary. And, I think we could do it in recompute_calibration_parameters by just passing in the dictionary of the action parameters to make if work for all signal analyzers.